### PR TITLE
BuyAudio Recovery Flow Fast Follows

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

#2043 Fast follows from PR comments

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-buy-recovery-ff

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-buy-recovery-ff

File: packages/web/src/services/audius-backend/BuyAudio.ts [hunk 1]
```diff
@@ -201,6 +201,50 @@ export const pollForSolBalanceChange = async ({
...
+  let retries = 0
+  while (transaction === initialTransaction && retries++ < maxRetryCount) {
+    console.debug(
+      `Polling wallet ${rootAccount.toString()} for new transaction.... [${retries}/${maxRetryCount}]`
+    )
+    await delay(retryDelayMs)
+    const transactions = await connection.getSignaturesForAddress(rootAccount, {
+      limit: 1
+    })
+    transaction = transactions[0].signature
```
Comment (sddioulde) [inline packages/web/src/services/audius-backend/BuyAudio.ts:239]: same deal with potential empty array error here

**Review by sddioulde — COMMENTED** [review]
looks most good, left comments

File: packages/web/src/store/application/ui/buy-audio/sagas.ts [hunk 2]
```diff
@@ -525,12 +526,20 @@ function* purchaseStep({
...
     'finalized'
   )
+  const initialTransactions = yield* call(
+    [connection, connection.getSignaturesForAddress],
+    rootAccount.publicKey,
+    {
+      limit: 1
+    }
+  )
+  const initialTransaction = initialTransactions[0].signature
```
Comment (sddioulde) [inline packages/web/src/store/application/ui/buy-audio/sagas.ts:542]: sorry didnt catch this one. but potentially empty array here too

**Review by sddioulde — APPROVED** [review]
left another comment. shoulda caught in prev review

File: packages/web/src/store/application/ui/buy-audio/sagas.ts (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/store/application/ui/buy-audio/sagas.ts:542]: oops! i should have seen that, my b

**Pull request merged**